### PR TITLE
Illustrated that links between gallery examples are possible.

### PR DESCRIPTION
**Pull request by emmanuelle**
Links between gallery examples are a useful feature. I added such a link to one of the ``sin_func`` gallery examples. Maybe this feature could be documented in ``advanced_configuration.rst`` as well?

### CODE REVIEW

File: examples/sin_func/plot_sin.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
     x \\rightarrow \\sin(x)
 
+.. seealso::
+    :ref:`sphx_glr_auto_examples_sin_func_plot_sin_black_background.py` for a
+    fancier plot
```
Comment (GaelVaroquaux) [inline examples/sin_func/plot_sin.py]: Maybe here you should add a text saying why this is there, for instance

"Note that sphinx-gallery automatically creates labels for each example from it's filename. You can thus refer to it from any part of the documentation, including from other examples, as here".

**Comment by GaelVaroquaux** [comment]
+1 for merge although I would add the sentence above :).

I don't think that "advanced configuration" is the part of the narrative documentation where this should be added: it's not a configuration issue. That said, I agree that mentioning this in the narrative documentation would be great.

**Comment by Titan-C** [comment]
I would also agree on including the same note on the readme.rst, maybe on the `The structure of the examples folder` section. Or indeed as a short note in the in the `features` section of the index.rst file

**Comment by GaelVaroquaux** [comment]
I don't think that it should be in the readme: the readme should focus on
the big-picture.

**Comment by agramfort** [comment]
thx @emmanuelle !

**Pull request merged**